core/domain/feedback_domain.py

```diff
@@ -29,7 +29,7 @@ class FeedbackThreadDict(TypedDict):
 
     last_updated_msecs: float
     original_author_id: str
-    state_name: str
+    state_name: Optional[str]
```

Comment:
Here we are providing none ( 4th argument ) value :
https://github.com/oppia/oppia/blob/5c6c538ec98b9de3f3d95c1222d66bd2a31bda2a/core/domain/feedback_services.py#L675

Reply:
But what does this mean conceptually? Do we even use it somewhere?

Reply:
While converting `GeneralFeedbackThreadModel` to `FeedbackThread` domain object we are providing None to state_name because `GeneralFeedbackThreadModel` does not contain state_name field. Also, I didn't find any case where we are using this state_name but in the frontend state_name filed is defined:
https://github.com/oppia/oppia/blob/135e85b4419bd022b9da28718402697b28362262/core/templates/domain/feedback_thread/FeedbackThreadObjectFactory.ts#L33
Can you please take a look, also please correct me if I'm missing something!!

Reply:
Is that the only place where "state_name" is used? From my quick look it seems that the field is not used anywhere and thus can be removed.

Reply:
For the backend, it is not used ( at least I didn't find any case ). but if we see the front-end dict then there we are expecting `state_name`.
https://github.com/oppia/oppia/blob/135e85b4419bd022b9da28718402697b28362262/core/templates/domain/feedback_thread/FeedbackThreadObjectFactory.ts#L26

Reply:
I'm not sure about these changes because even if we removed `state_name` from here then in front_end we are still expecting `state_name` from the dictionary that was passed from the backend. 

My Opinion ~ If we want to remove `state_name` then we have to do it for the front_end and backend at the same time. Maybe I'm wrong please correct me if I'm wrong!!

Reply:
Yep it needs to be removed form BE and FE at once. I think this can be done in a separate PR.